app/controllers/notes_controller.rb

```diff
@@ -1,12 +1,31 @@
 class NotesController < ApplicationController
+  before_action :find_note, only: %i[edit update]
+  before_action :find_volunteer
+
   def create
-    volunteer = Volunteer.find(params[:volunteer_id])
-    volunteer.notes.create(note_params)
-    redirect_to edit_volunteer_path(volunteer)
+    @volunteer.notes.create(note_params)
```

Comment:
We are missing an `authorize` somewhere but that may be covered in a different ticket